salt/iiif/init.sls

```diff
@@ -0,0 +1,224 @@
+get-loris:
+    docker_image.present:
+        - name: elifesciences/loris:latest
+        - require:
+            - docker-ready
+
+
+# loris user is deprecated in favour of www-data
+
+{% set loris_user = pillar.elife.webserver.username %}
+
+loris-user:
+    user.present: 
+        - name: {{ loris_user }}
+        #- uid: ... # on vagrant this is 1003, everywhere else it's 1002. www-data is consistently uid 33
+        #- shell: /sbin/false
+        #- home: /home/loris
+        #- createhome: False
+
+# directories the container will have mounted
+
+loris-tmp-directory:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/tmp
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - dir_mode: 755
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+loris-cache-general:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/cache-general
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - dir_mode: 755
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+loris-cache-resolver:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/cache-resolver
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+# empty folder that can be synced over the caches to clean them
+loris-cache-blank:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/blank
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - dir_mode: 755
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+
+# Docker needs to bind certain configuration files between host and container
+
+
+loris-dir:
+    file.directory:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/loris
+
+loris-config:
+    file.managed:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/loris/loris2.conf
+        - source: salt://iiif/config/opt-loris-loris2.conf
+        - template: jinja
+        - makedirs: True
+        - require:
+            - loris-dir
+
+# required by newrelic-python.sls as it uses the 'newrelic' installed in the venv to generate the licence file
```

Comment:
argh

Reply:
Not the end of the world. I am able to re-use the whole newrelic-python state file which is a good trade-off